### ISSUE TITLE
TypeError: a bytes-like object is required, not 'str' in file_uploader

### ISSUE THREAD

**Issue by abhishekupadhyaya** [forseti-security/forseti-security#3034]
Hi,

We are using `release-2.18.0`. We found a following error in `upload_json` method of [`file_uploader.py`](https://github.com/forseti-security/forseti-security/blob/release-2.18.0/google/cloud/forseti/common/util/file_uploader.py):
```
TypeError: a bytes-like object is required, not 'str'
```
Line 38 in file_uploader.py should instead be:
```
tmp_data.write(parser.json_stringify(data).encode('utf-8'))
```
This can also be modified in `json_stringify` method in `parser.py`

Thank you,
Abhishek

**Comment by blueandgold** [comment]
@abhishekupadhyaya Can you give us the entire stacktrace so that we can see what is the caller the caused this error?

**Comment by marciosian** [comment]
I'm facing the same issue. v2.18  

2019-08-13 10:05:34,283 ERROR [forseti-security][2.18.0] google.cloud.forseti.common.util.file_uploader(upload_json): Unable to upload json document to bucket gs://forseti-server-***/scanner_violations/violations.ke_version_violations.1565690410195284.20190813T100534Z.json:
[{'id': 859, 'created_at_datetime': '2019-08-13T10:05:32Z', ***
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/forseti_security-2.18.0-py3.6.egg/google/cloud/forseti/common/util/file_uploader.py", line 38, in upload_json
    tmp_data.write(parser.json_stringify(data))
  File "/usr/lib/python3.6/tempfile.py", line 624, in func_wrapper
    return func(*args, **kwargs)
TypeError: a bytes-like object is required, not 'str'

If i switch to csv it works well.

**Comment by kevensen** [comment]
@marciosian Thank you for providing the stack trace.  I'll investigate.

**Comment by kevensen** [comment]
I have reproduced the issue.  Continuing to investigate.

**Comment by kevensen** [comment]
@abhishekupadhyaya @marciosian The culprit is indeed line 38 of file uploader.

https://github.com/forseti-security/forseti-security/blob/b54faf68d869842e8a43472ff980e28e2ce8d3c6/google/cloud/forseti/common/util/file_uploader.py#L36-L39

In file_uploader.py, could you please change line 38 to be:

```python
tmp_data.write(parser.json_stringify(data).encode('utf-8'))
```

Then run the below to apply the changes:
```bash
# install inside the /home/ubuntu/forseti-security folder within the VM
sudo python3 setup.py 
sudo systemctl restart forseti
```

We'll put this in our backlog to fix.

**Comment by marciosian** [comment]
It works!

**Comment by kevensen** [comment]
@marciosian Great!  This fix will be included in v2.19.1
